src/engagement.js

```diff
@@ -0,0 +1,154 @@
+          <div class="show-description-1">
+            <p>Type: ${show.type}</p>
+            <p>Language: ${show.language}</p>
+          </div>
+          <div class="show-description-2">
+            <p>Status: ${show.status}</p>
+            <p>Premiered: ${show.premiered}</p>
+          </div>
+          <div class="comments-display">
+            ${commentBlock}
+          </div>
+          <div class="comment-create">
+            <h3>Add a comment</h3>
+            <input name="username" placeholder="Your name" />
+            <textarea name="insights" rows="6" placeholder="Your Thoughts"></textarea>
+            <span type="button" comment-id="${show.id}" class="material-icons-outlined">comment</span>
+          </div>
+        </div>
+      </div>`);
+  document.querySelector('main').appendChild(popup);
+  const closeButton = document.getElementsByClassName('close-popup')[0];
+  closeButton.addEventListener('click', () => {
+    document.querySelector('.popup').remove();
+  });
+
+  const commentsDisplay = document.querySelectorAll(`[comment-id="${show.id}"]`)[0]
+    .parentElement.previousElementSibling;
+
+  if (comments.length > 0) {
+    commentBlock += '<h3>Comments</h3>';
+    comments.forEach((comment) => {
+      const date = comment.creation_date.split('-');
+      const dateFormated = `${date[1]}/${date[2]}/${date[0]}`;
+      commentBlock += `<p>${dateFormated} ${comment.username}: ${comment.comment}</p>`;
+    });
+  }
+  commentsDisplay.insertAdjacentHTML('beforeend', commentBlock);
+
+  updateCommentTitle(show.id);
+
+  const commentButton = document.querySelectorAll(`[comment-id="${show.id}"]`)[0];
+  commentButton.addEventListener('click', async (e) => {
+    const commentObject = {
+      item_id: Number(e.target.getAttribute('comment-id')),
+      username: commentButton.previousElementSibling.previousElementSibling.value,
+      comment: commentButton.previousElementSibling.value,
+    };
+
+    const result = await createComment(commentObject);
+
+    if (result === 201) {
+      const comments = await getComments(showId);
+      const lastComment = comments[comments.length - 1];
+      const commentsDisplay = document.querySelectorAll(`[comment-id="${show.id}"]`)[0]
+        .parentElement.previousElementSibling;
+      const date = lastComment.creation_date.split('-');
+      const dateFormated = `${date[1]}/${date[2]}/${date[0]}`;
+
+      if (comments.length === 1) {
+        commentsDisplay.insertAdjacentHTML('beforeend', `
+            <h3>Comments</h3>
+            <p>${dateFormated} ${lastComment.username}: ${lastComment.comment}</p>
+          `);
+      } else {
+        commentsDisplay.insertAdjacentHTML('beforeend', `
+            <p>${dateFormated} ${lastComment.username}: ${lastComment.comment}</p>
+          `);
+      }
+      updateCommentTitle(show.id);
+    }
+  });
+};
+
+const displayPopup = async (id) => {
+  const show = await pullId(id);
+  genPopupContent(show);
+};
```

Comment:
The file is a bit long, you could make use of modules and divide in order to make it easier to perform!